mongodump/prepare.go

```diff
@@ -194,9 +196,26 @@ func (dump *MongoDump) outputPath(dbName, colName string) string {
 		root = dump.OutputOptions.Out
 	}
 
-	return filepath.Join(root, dbName, util.EscapeCollectionName(colName))
+	// Encode a new output path for collection names that would result in a file name greater
+	// than 255 bytes long. This includes the longest possible file extension: .metadata.json.gz
+	// The new format is <truncated-url-encoded-collection-name>%24<collection-name-hash-base64>
+	// where %24 represents a $ symbol delimiter (e.g. aVeryVery...VeryLongName%24oPpXMQ...).
+	if len(colName) > 238 {
+		colNameTruncated := util.EscapeCollectionName(colName)[:208]
+		delimiter := util.EscapeCollectionName("$")
+		colNameHashBytes := sha1.Sum([]byte(colName))
+		colNameHashBase64 := base64.RawURLEncoding.EncodeToString(colNameHashBytes[:])
+
+		// First 208 bytes of col name + 3 bytes delimiter + 27 bytes base64 hash = 238 bytes.
+		colName = colNameTruncated + delimiter + colNameHashBase64
+	} else {
+		colName = util.EscapeCollectionName(colName)
+	}
+
+	return filepath.Join(root, dbName, colName)
 }
 
+// Unused.
```

Comment:
Did you add this comment because the `checkStringForPathSeparator` function is not used anywhere? If it is not used anywhere, should we delete it?

Reply:
Right, it's dead code now. I can remove it if Tim agrees we wouldn't need it for anything else.

Reply:
Yep let's delete it!